src/server/game/Server/Packets/MovementPackets.h

```diff
@@ -710,6 +710,35 @@ namespace WorldPackets
 
             uint32 Ticks = 0;
         };
+
+        class SetAdvFlyingSpeed final : public ServerPacket
+        {
+        public:
+            SetAdvFlyingSpeed(OpcodeServer opcode, uint32 sequenceIndex, float speed, Optional<float> maxSpeed = {}) : ServerPacket(opcode, 4 + 4 + 4)
+            {
+                SequenceIndex = sequenceIndex;
+                Speed = speed;
+                MaxSpeed = maxSpeed;
+            }
+
+            WorldPacket const* Write() override;
+
+            uint32 SequenceIndex = 1;
+            float Speed = .0f;
+            Optional<float> MaxSpeed;
```

Comment:
I want this split up into two classes `SetAdvFlyingSpeed` and `SetAdvFlyingMinMaxSpeeds` since this isn't a shared reading function in client and using Optional in packets implies that it is